access/errors.go

```diff
@@ -98,3 +98,12 @@ type InvalidTxRateLimitedError struct {
 func (e InvalidTxRateLimitedError) Error() string {
 	return fmt.Sprintf("transaction rate limited for payer (%s)", e.Payer)
 }
+
+type InsufficientBalanceError struct {
+	Payer           flow.Address
+	RequiredBalance uint64
```

Comment:
This will print `1.0` as `100000000`, which isn't very user friendly. How about taking a `cadence.UFix64` here, then you can use `e.RequiredBalance.String()` to convert it into a decimal string value:
https://github.com/onflow/cadence/blob/be2d8ffd8ebb5e0c2e846fd0b42c737912d374bc/values.go#L1382-L1384